src/runtime/pay-flow.js

```diff
@@ -166,26 +166,33 @@ export function validatePayResponse(port) {
  * @return {!SubscribeResponse}
  */
 export function parseSubscriptionResponse(data) {
+  let swgData = null;
   let raw = null;
   if (data) {
     if (typeof data == 'string') {
       raw = /** @type {string} */ (data);
     } else {
       // Assume it's a json object in the format:
-      // `{integratorClientCallbackData: "..."}`.
+      // `{integratorClientCallbackData: "..."}` or `{swgCallbackData: "..."}`.
       const json = /** @type {!Object} */ (data);
-      raw = json['integratorClientCallbackData'];
+      if ('swgCallbackData' in json) {
+        swgData = /** @type {!Object} */ (json['swgCallbackData']);
+      } else if ('integratorClientCallbackData' in json) {
+        raw = json['integratorClientCallbackData'];
+      }
     }
   }
-
-  let swgData = null;
-  if (raw) {
-    const parsed = parseJson(atob(raw));
-    swgData = parsed['swgCallbackData'];
+  if (raw && !swgData) {
+    raw = atob(raw);
+    if (raw) {
+      const parsed = parseJson(raw);
+      swgData = parsed['swgCallbackData'];
+    }
   }
   if (!swgData) {
     throw new Error('unexpected payment response');
   }
+  raw = JSON.stringify(/** @type {!JsonObject} */ (swgData));
   return new SubscribeResponse(
       raw,
```

Comment:
Return the json object instead of Json.stringify?
It's fine to do this in another pull request if needed

Reply:
By current format, `raw` is a string to simplify posting it to the 3p site.